src/transformers/models/codegen/tokenization_codegen.py

```diff
@@ -150,13 +152,17 @@ def __init__(
         pad_token=None,
         add_prefix_space=False,
         add_bos_token=False,
+        return_token_type_ids=False,
         **kwargs,
     ):
         bos_token = AddedToken(bos_token, special=True) if isinstance(bos_token, str) else bos_token
         eos_token = AddedToken(eos_token, special=True) if isinstance(eos_token, str) else eos_token
         unk_token = AddedToken(unk_token, special=True) if isinstance(unk_token, str) else unk_token
         pad_token = AddedToken(pad_token, special=True) if isinstance(pad_token, str) else pad_token
         self.add_bos_token = add_bos_token
+        self.return_token_type_ids = return_token_type_ids
+        if self.return_token_type_ids:
+            self.model_input_names.append("token_type_ids")
```

Comment:
if we always add it does it make a difference? It's my first time seeing this fix, so a bit suprised that we have to do that

Reply:
Yes. This actually controls whether the tokenizer returns `token_type_ids` or not in [here](https://github.com/huggingface/transformers/blob/8dd4ce6f2cea1316cb5bd5ccbd348310ced17856/src/transformers/tokenization_utils_base.py#L3494-L3495). The tokenizer doesn't return `token_type_ids` without this line.

Reply:
Sorry you are right! It's not really well made. I'll approve, but would be nice if we can fix self.model_iinput_names based on the value of `return_token_type_ids`. For now only the reverse is True.